### PR TITLE
feat(cli): add prof and tokio env config

### PR DESCRIPTION
**Pull request by skyzh**
Signed-off-by: Alex Chi <iskyzh@gmail.com>

I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Now we can use environment variable to control tokio, deadlock detection, and profile flame graph behavior. See docs for more information.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)

close https://github.com/singularity-data/risingwave/issues/1536
close https://github.com/singularity-data/risingwave/issues/3792

### CODE REVIEW

File: src/utils/runtime/src/lib.rs [hunk 1]
```diff
@@ -228,13 +231,75 @@ pub fn enable_parking_lot_deadlock_detection() {
...
 
-    if cfg!(debug_assertion) {
-        enable_parking_lot_deadlock_detection();
+    let mut builder = tokio::runtime::Builder::new_multi_thread();
+
+    if let Ok(worker_threads) = std::env::var("RW_WORKER_THREADS") {
+        let worker_threads = worker_threads.parse().unwrap();
+        tracing::info!("setting tokio worker threads to {}", worker_threads);
+        builder.worker_threads(worker_threads);
+    }
```
Comment (BugenZhao) [inline src/utils/runtime/src/lib.rs:286]: May use `env_var_is_true` for this.

https://github.com/singularity-data/risingwave/blob/218dc93a137060ca09bcd63a4fbdad20e81c540a/src/common/src/util/env_var.rs#L18

**Review by TennyZhuang — APPROVED** [review]

File: src/utils/runtime/src/lib.rs (hunk 1 above)
Comment (skyzh) [reply src/utils/runtime/src/lib.rs:286]: We need to tell the difference between "true", "false" and "not exist", so I think `env_var_is_true` is not a good fit here?

Comment (skyzh) [reply src/utils/runtime/src/lib.rs:286]: Also this crate doesn't depend on common, I'd like to keep crates away from critical path as possible 🤣

File: src/utils/runtime/src/lib.rs [hunk 2]
```diff
@@ -228,13 +231,75 @@ pub fn enable_parking_lot_deadlock_detection() {
...
-/// * Set panic hook to abort the whole process.
-pub fn oneshot_common() {
+/// Currently, the following env variables will be read:
+///
+/// * `RW_WORKER_THREADS`: number of tokio worker threads. If not set, it will use tokio's default
+///   config (equivalent to CPU cores).
+/// * `RW_DEADLOCK_DETECTION`: whether to enable deadlock detection. If not set, will enable in
+///   debug mode, and disable in release mode.
+/// * `RW_PROFILE_PATH`: the path to generate flamegraph. If set, then profiling is automatically
+///   enabled.
```
Comment (BowenXiao1999) [inline src/utils/runtime/src/lib.rs:273]: How we expect devs to know this? Only be looking at the code doc? 🤔 
Will we provide a config file?

**Review by BowenXiao1999 — APPROVED** [review]

File: src/utils/runtime/src/lib.rs (hunk 2 above)
Comment (skyzh) [reply src/utils/runtime/src/lib.rs:273]: I think this is only helpful for developers, so it won't be added in user docs. For developers, currently I have no idea how to gain more visibility on these parameters. Let's keep it in code doc for now.

**Pull request merged**